osgtest/tests/test_11_condor_cron.py

```diff
@@ -1,24 +1,31 @@
 import os
 import osgtest.library.core as core
 import osgtest.library.osgunittest as osgunittest
+import osgtest.library.service as service
+import time
 import unittest
 
 class TestStartCondorCron(osgunittest.OSGTestCase):
-
     def test_01_start_condor_cron(self):
-        core.config['condor-cron.lockfile'] = '/var/lock/subsys/condor-cron'
         core.state['condor-cron.started-service'] = False
         core.state['condor-cron.running-service'] = False
 
         core.skip_ok_unless_installed('condor-cron')
-        if os.path.exists(core.config['condor-cron.lockfile']):
+        if service.is_running('condor-cron'):
             core.state['condor-cron.running-service'] = True
             self.skip_ok('already running')
 
-        command = ('service', 'condor-cron', 'start')
-        stdout, _, fail = core.check_system(command, 'Start Condor-Cron')
-        self.assert_(stdout.find('error') == -1, fail)
-        self.assert_(os.path.exists(core.config['condor-cron.lockfile']),
-                     'Condor-Cron run lock file missing')
+        if core.el_release() < 7:
+            command = ('service', 'condor-cron', 'start')
+            stdout, _, fail = core.check_system(command, 'Start Condor-Cron')
+            self.assert_(stdout.find('error') == -1, fail)
+        else:
+            core.check_system(('systemctl', 'start', 'condor-cron'), 'Start Condor-Cron')
+
+        # Give condor-cron time to start up
+        time.sleep(5)
+
+        self.assert_(service.is_running('condor-cron'), "Condor-Cron is not running")
```

Comment:
`service.is_running()` turns out to work here because the lockfile is the same between the init script and systemd but this isn't always the case (HTCondor-CE comes to mind). If the lockfile names are different, and the init `status()` function checks for the lockfile, it will always fail since the lockfile exists elsewhere. We should update `service.is_running()` to use `systemctl is-active` .